service/instance.py

```diff
@@ -1319,57 +1321,111 @@ def user_delete_security_group(core_identity):
 
 def security_group_init(core_identity, max_attempts=3):
     has_secret = core_identity.get_credential('secret') is not None
+    security_group_name = core_identity.provider.get_config("network", "security_group_name", "MOC-GIJI")
     if has_secret:
         return admin_security_group_init(core_identity)
-    return user_security_group_init(core_identity)
+    return user_security_group_init(core_identity, security_group_name=security_group_name)
 
 
-def user_security_group_init(core_identity, security_group_name='default'):
-    # Rules can come from the provider _or_ from settings _otherwise_ empty-list
-    rules = core_identity.provider.get_config('network', 'default_security_rules', getattr(settings, 'DEFAULT_RULES', []))
-    driver = get_cached_driver(identity=core_identity)
-    lc_driver = driver._connection
-    security_group = get_or_create_security_group(lc_driver, security_group_name)
-    set_security_group_rules(lc_driver, security_group, rules)
+def user_security_group_init(core_identity, security_group_name):
+    network_driver = _to_network_driver(core_identity)
+    user_neutron = network_driver.neutron
+    extended_default_rules = _get_default_rules()
+    user_security_group_rules = core_identity.provider.get_config('network', 'user_security_rules', extended_default_rules)
+    security_group = get_or_create_security_group(security_group_name, user_neutron)
+    neutron_set_security_group_rules(security_group_name, user_security_group_rules, user_neutron)
     return security_group
 
 
-def set_security_group_rules(lc_driver, security_group, rules):
-    for rule_tuple in rules:
-        if len(rule_tuple) == 3:
-            (ip_protocol, from_port, to_port) = rule_tuple
-            cidr = None
-        elif len(rule_tuple) == 4:
-            (ip_protocol, from_port, to_port, cidr) = rule_tuple
-        else:
-            raise Exception("Invalid DEFAULT_RULES contain a rule, %s, which does not match the expected format" % rule_tuple)
-
+def _get_default_rules():
```

Comment:
Thanks Laura, good suggestion. It was Rado and I decided to only put port 22 there. But both of us agree that ping should also be there. I will send out another PR to change it.